Add support for slash commands

V13Axel opened the pull request:
This addresses #227 and allows the use of Slack slash commands, as below.

Simply prepend `/slack slash` to whatever you'd run via the official clients, and your command will be run. Example:

`/command arg1 arg2 arg3` simply becomes `/slack slash /command arg1 arg2 arg3`.

I've found it helpful to create an alias of `/command` => `/slack slash command` via weechat for the ones I use.

tfheen commented on wee_slack.py:
```diff
@@ -1308,6 +1308,30 @@ def command_markread(current_buffer, args):
+    aargs = args.split(None, 1)
+
+    if len(aargs) == 0:
+        # XXX: whine
```
We should offer a useful error in this case.

tfheen commented on wee_slack.py:
```diff
@@ -1308,6 +1308,30 @@ def command_markread(current_buffer, args):
+    domain = current_domain_name()
+
+    if servers.find(domain).channels.find(channel):
+        channel_identifier = servers.find(domain).channels.find(channel).identifier
```
What happens if we don't find the channel?

V13Axel commented:
@tfheen How's that look? Cleaned things up a bit and addressed both of your concerns (As well as an error for just typing `/slack slash /command` with no args)

tfheen commented on wee_slack.py:
```diff
@@ -1308,6 +1308,37 @@ def command_markread(current_buffer, args):
+    domain = current_domain_name()
+
+    if args is None:
+        server.buffer_prnt("Usage: /slack slash [/someslashcommand].")
```
This should probably read `"Usage: /slack slash /someslashcommand [arguments...]"`

tfheen commented on wee_slack.py:
```diff
@@ -1308,6 +1308,37 @@ def command_markread(current_buffer, args):
+    aargs = args.split(None, 1)
+
+    command = aargs[0]
+    if 1 < len(aargs):
```
`text = args[1] if len(args > 1) else ""`

V13Axel commented:
Whoot, +1 for @rawdigits double-checking my testing of it and merging it in :wink:

l2ol33rt commented:
@tfheen any update on testing this?

I'm running it off @V13Axel's `master` branch and it's awesome!

V13Axel commented on README.md:
```diff
@@ -32,6 +32,7 @@ Features
   * Away/back status handling
   * Expands/shows metadata for things like tweets/links
   * Displays edited messages (slack.com irc mode currently doesn't show these)
+  * Slash commands (including custom ones!)
```
Heh. I had forgotten that existed, will do.

kenyonj replied:
:+1: Excited to get this in!
